utils/build-script

```diff
@@ -848,14 +848,21 @@ class BuildScriptInvocation(object):
         return options
 
     def compute_product_classes(self):
-        """compute_product_classes() -> (list, list)
-
-        Compute the list first of all build-script-impl products and then all
-        non-build-script-impl products. It is assumed that concatenating the two
-        lists together will result in a valid dependency graph for the
-        compilation.
+        """compute_product_classes() -> (list, list, list)
 
+        Compute the list first of all pre-build-script-impl products, then all
+        build-script-impl products and then all non-build-script-impl products.
+        It is assumed that concatenating the three lists together will result in a
+        valid dependency graph for the compilation.
         """
+        before_impl_product_classes = []
+        # If --skip-early-swift-driver is passed in, swift will be built
+        # as usual, but relying on its own C++-based (Legacy) driver.
+        # Otherwise, we build an "early" swift-driver using the host
+        # toolchain, which the later-built compiler will forward
```

Comment:
I would add here that we do this only if we find swiftc in the host toolchain.

Reply:
Clarified. Thanks.